src/ttsim/plot/dag/tt.py

```diff
@@ -113,46 +135,80 @@ def tt(
-        policy_environment=environment,
-        labels={"processed_data_columns": qnames_policy_inputs},
-        tt_targets={"qname": qnames_to_plot},
-        backend="numpy",
+    complete_tt_dag_and_specialized_environment = main(
+        main_targets=[
+            MainTarget.specialized_environment_for_plotting_and_templates.complete_tt_dag,
+            MainTarget.specialized_environment_for_plotting_and_templates.without_tree_logic_and_with_derived_functions,
+        ],
+        policy_date_str=policy_date_str,
+        orig_policy_objects=(
+            orig_policy_objects if orig_policy_objects else OrigPolicyObjects(root=root)
+        ),
+        policy_environment=policy_environment,
+        tt_targets=tt_targets,
+        input_data=input_data,
+        processed_data=processed_data,
+        labels=labels,
+        backend=backend,
+        include_fail_nodes=include_fail_nodes,
+        include_warn_nodes=include_warn_nodes,
+    )
+    complete_tt_dag = complete_tt_dag_and_specialized_environment[
+        "specialized_environment_for_plotting_and_templates"
+    ]["complete_tt_dag"]
+    without_tree_logic_and_with_derived_functions = (
+        complete_tt_dag_and_specialized_environment[
+            "specialized_environment_for_plotting_and_templates"
+        ]["without_tree_logic_and_with_derived_functions"]
     )
 
-    all_nodes = convert_all_nodes_to_callables(env)
-
-    complete_dag = dags.create_dag(functions=all_nodes, targets=qnames_to_plot)
-
-    if node_selector is None:
-        selected_dag = complete_dag
+    if not selection_type:
+        selected_dag = complete_tt_dag
     else:
-        selected_dag = _create_dag_with_selected_nodes(
-            complete_dag=complete_dag,
-            node_selector=node_selector,
+        tt_targets_qnames = _get_tt_targets_qnames(tt_targets)
+        _fail_if_tt_targets_not_provided(tt_targets_qnames)
+        selected_dag = select_nodes_from_dag(
+            complete_tt_dag=complete_tt_dag,
+            tt_targets_qnames=tt_targets_qnames,
+            selection_type=selection_type,
+            selection_depth=selection_depth,
         )
 
     if not include_params:
-        selected_dag.remove_nodes_from(
-            [qn for qn, v in env.items() if isinstance(v, (ParamObject, ParamFunction))]
-        )
+        qnames_params: set[str] = set()
+        for qn, v in without_tree_logic_and_with_derived_functions.items():
+            if isinstance(v, (ParamObject, ParamFunction)):
+                qnames_params.add(qn)
+            elif (
+                isinstance(v, TimeConversionFunction)
+                and hasattr(v, "source")
+                and v.source in without_tree_logic_and_with_derived_functions
+                and isinstance(
+                    without_tree_logic_and_with_derived_functions[v.source],
+                    (ParamObject, ParamFunction),
+                )
+            ):
+                # Also add time-converted params
```

Comment:
```suggestion
                # Also schedule time-converted params for removal.
```
We add to the set, but the goal is different...